provider/clickhouse.go

```diff
@@ -577,33 +643,73 @@ func (it *clickHouseTableIterator) Close() error {
 }
 
 func (store *clickHouseOfflineStore) RegisterResourceFromSourceTable(id ResourceID, schema ResourceSchema, opts ...ResourceOption) (OfflineTable, error) {
+	logger := store.logger.With("id", id, "schema", schema, "opts", opts)
+
 	if len(opts) > 0 {
-		return nil, fferr.NewInternalErrorf("ClickHouse does not support resource options")
+		errorMsg := "ClickHouse does not support resource options"
+		logger.Errorw(errorMsg)
+		return nil, fferr.NewInternalErrorf(errorMsg)
 	}
+
 	if err := id.check(Feature, Label); err != nil {
+		logger.Errorw("invalid resource type, must be feature or label")
 		return nil, err
 	}
-	if exists, err := store.tableExists(id); err != nil {
+
+	chLocation, err := clickhouse.NewLocation(schema.SourceTable)
+	if err != nil {
+		logger.Errorw("source table location is not ClickHouse compatible")
+		return nil, fferr.NewExecutionError(pt.ClickHouseOffline.String(), err)
+	}
+
+	expectedColumns, err := schema.ToColumnStringSet(id.Type)
+	if err != nil {
+		logger.Errorw("failed to get expected columns", "error", err)
 		return nil, err
-	} else if exists {
-		return nil, fferr.NewDatasetAlreadyExistsError(id.Name, id.Variant, nil)
 	}
-	if schema.Entity == "" || schema.Value == "" {
-		return nil, fferr.NewInvalidArgumentError(fmt.Errorf("non-empty entity and value columns required"))
+
+	// TODO: This is done because the method on the query interface expects a
+	// FullyQualifiedObject. That should ideally be refactored, as well as other
+	// general Location refactorings. Once that is done, this can be removed.
+	fullyQualifiedObject := pl.FullyQualifiedObject{
+		Database: chLocation.GetDatabase(),
+		Schema:   "",
+		Table:    chLocation.GetTable(),
 	}
-	tableName, err := store.getResourceTableName(id)
+
+	query, err := store.query.resourceTableColumns(fullyQualifiedObject)
 	if err != nil {
+		logger.Errorw("error creating resourceTableColumns query", "error", err)
 		return nil, err
 	}
 
-	if err := store.query.registerResources(store.db, tableName, schema); err != nil {
+	rows, err := store.db.Query(query)
```

Comment:
Can you make this a new method?

ValidateColumnsExist? or something like that?